function.go

```diff
@@ -433,6 +366,50 @@ func assertEmptyRoot(path string) (err error) {
 	return
 }
 
+// ImageName returns a full image name (OCI container tag) for the
+// Function based off of the Function's `Registry` member plus `Name`.
+// Used to calculate the final value for .Image when none is provided
+// explicitly.
+//
+// form:    [registry]/[user]/[function]:latest
+// example: quay.io/alice/my.function.name:latest
+//
+// Also nested namespaces should be supported:
+// form:    [registry]/[parent]/[user]/[function]:latest
+// example: quay.io/project/alice/my.function.name:latest
+//
+// Registry values which only contain a single token are presumed to
+// indicate the namespace at the default registry.
+func (f Function) ImageName() (image string, err error) {
```

Comment:
This would probably better be `f.Image()` with logic for derivation inbuilt, but that's a larger proposition since it affects the struct's serialization.